src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -2978,7 +2978,22 @@ bool G1PrintRegionLivenessInfoClosure::do_heap_region(HeapRegion* r) {
   _total_strong_code_roots_bytes += strong_code_roots_bytes;
 
   // Print a line for this particular region.
-  log_trace(gc, liveness)(G1PPRL_LINE_PREFIX
+  if(gc_eff < 0) {
```

Comment:
I would prefer instead of the code duplication, use a `%s` format specifier for the efficiency, and a `FormatBuffer` to format the actual string into it. This should result in much shorter code.

Reply:
Yes, I absolutely agree! Thanks!